src/QAT/qat/purr/compiler/runtime.py

```diff
@@ -256,7 +255,11 @@ def execute_instructions(
     active_runtime = get_runtime(hardware)
 
     active_runtime.run_quantum_executable(executable_blocks)
+
+    results, engine_call_duration_metrics = active_runtime.execute(instructions, results_format, repeats)
```

Comment:
Find where `active_runtime.compilation_metrics` is used and remove it from the runtime, instead returning it in the new metrics object. It was always going to be moved anyway, since runtimes persist beyond a single running instance. 

You can then turn all of this into just:

```
return active_runtime.execute(...)
```